### PR TITLE
APIS-4825 - Adding ApiContext value class

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

File: app/services/ApplicationService.scala [hunk 1]
```diff
@@ -99,7 +99,7 @@ class ApplicationService @Inject()(sandboxApplicationConnector: SandboxApplicati
 
       val apps = par.applications.map(ar => {
         val filteredSubs = subscriptions.filter(_.applications.exists(_ == ar.id.toString))
-          .map(sub => SubscriptionNameAndVersion(sub.apiIdentifier.context, sub.apiIdentifier.version)).sortBy(sub => sub.name)
+        .map(sub => SubscriptionNameAndVersion(sub.apiIdentifier.context.value, sub.apiIdentifier.version)).sortBy(sub => sub.name)
```
Comment (AndySpaven) [inline app/services/ApplicationService.scala:102]: Not typed?

File: app/utils/SubscriptionEnhancer.scala [hunk 2]
```diff
@@ -25,13 +25,13 @@ trait SubscriptionEnhancer {
               definitions: Seq[APIDefinition]): PaginatedDetailedSubscribedApplicationResponse = {
     val apps = appResponse.applications.map { ar =>
       val details = ar.subscriptions.map(sub =>
-        SubscriptionDetails(definitions.find(_.context == sub.name) match {
+        SubscriptionDetails(definitions.find(_.context.value == sub.name) match {
```
Comment (AndySpaven) [inline app/utils/SubscriptionEnhancer.scala:28]: Is this not that Sub Details should use the type?

File: it/ApiStatusMappingIntegrationSpec.scala [hunk 3]
```diff
@@ -37,6 +38,7 @@ class ApiStatusMappingIntegrationSpec extends UnitSpec with Matchers with GuiceO
   trait Setup {
     implicit val hc = HeaderCarrier()
     val connector = app.injector.instanceOf[ProductionApiDefinitionConnector]
+    val apiContext = ApiContext("dummy-api")
```
Comment (AndySpaven) [inline it/ApiStatusMappingIntegrationSpec.scala:41]: .random ?

File: test/connectors/SubscriptionFieldsConnectorSpec.scala [hunk 4]
```diff
@@ -52,6 +54,8 @@ class SubscriptionFieldsConnectorSpec extends UnitSpec with ScalaFutures with Mo
 
   def subscriptionFieldsBaseUrl(clientId: ClientId) = s"$urlPrefix/application/${clientId.value}"
 
+  def urlEncode(str: String, encoding: String = "UTF-8") = encode(str, encoding)
```
Comment (AndySpaven) [inline test/connectors/SubscriptionFieldsConnectorSpec.scala:57]: Should we add urlEncode to apiContext and others as a way of ensuring it's done right?  Would save the need for .value.

File: test/connectors/SubscriptionFieldsConnectorSpec.scala [hunk 5]
```diff
@@ -111,7 +116,7 @@ class SubscriptionFieldsConnectorSpec extends UnitSpec with ScalaFutures with Mo
 
     val prefetchedDefinitions = Map(apiIdentifier -> Seq(subscriptionDefinition))
 
-    val getUrl = s"${subscriptionFieldsBaseUrl(clientId)}/context/$apiContext/version/$apiVersion"
+    val getUrl = s"${subscriptionFieldsBaseUrl(clientId)}/context/${urlEncode(apiContext.value)}/version/$apiVersion"
```
Comment (AndySpaven) [inline test/connectors/SubscriptionFieldsConnectorSpec.scala:119]: See above comment....

**Review by adampridmore — APPROVED** [review]

**Pull request merged**